richgo.go

```diff
@@ -30,27 +32,25 @@ func main() {
 	case 1:
 		cmd = exec.Command("go")
 	default:
-		cmd = exec.Command("go", os.Args[1:]...)
-		// select a wrapper with subcommand
-		if f, ok := lps[os.Args[1]]; ok {
-			factory = f
+		// This is a bit of a special case. Somebody is already
+		// running `go test` for us, and just wants us to prettify the
+		// output.
+		if os.Args[1] == testFilterCmd {
+			cmd = exec.Command("cat", "-")
```

Comment:
On the fence as to whether this is hacky or elegant. It requires the least code change, but feels a bit silly to spawn a `cat` process just to read from stdin. Then again, it fits naturally with the existing code...